doc/nrf/ug_app_dev.rst

```diff
@@ -56,3 +56,46 @@ In order to define your own board, you can use the following Zephyr guides as re
 
 * :ref:`custom_board_definition` is a guide to adding your own custom board to the Zephyr build system.
 * :ref:`board_porting_guide` is a complete guide to porting Zephyr to your own board.
+
+Memory footprint optimization
+*****************************
+
+When developing an application, ROM and RAM footprint are important factors, especially when the firmware runs on the most resource-constrained devices e.g. nRF52810 or nRF52811.
+If you want your application to use the minimum required resources, follow the steps below and tune the |NCS| configuration parameters reducing the memory footprint:
+
+# Follow the basic hints for Zephyr :ref:`zephyr:optimizations`
+# Analyze stack usage in each thread of your application using :ref:`zephyr:thread_analyzer` and reduce stack sizes where possible.
+# Limit or disable debugging features such as logging, asserts, etc.
+# Go through each component/subsystem and turn off features your application does not use.
+
+Below, you can find some details on the optimization of selected subsystems.
+
+Bluetooth
+=========
+
+* Disable features your application does not use e.g.:
+  * Data Length Update,
+  * Extended Advertising,
+  * PHY Update,
+  * Security Manager Protocol (if no encryption and authentication is needed),
+  * GATT Caching,
+  * GATT Service Changed, etc.
+
+* Reduce the stack sizes of the Bluetooth internal threads where possible (based on the output of the Thread Analyzer):
```

Comment:
We could refer to the documentation of the thread analyzer here:
https://docs.zephyrproject.org/2.3.0/guides/debugging/thread-analyzer.html

Reply:
It is linked in line 67, but I added a link here too.